src/relion/zocalo/images_service_plugin.py

```diff
@@ -161,3 +161,54 @@ def picked_particles(plugin_params):
         extra={"image-processing-time": timing},
     )
     return outfile
+
+
+def mrc_central_slice(plugin_params):
+    filename = plugin_params.parameters("file")
+    if not filename or filename == "None":
+        logger.error("Skipping mrc to jpeg conversion: filename not specified")
+        return False
+    filepath = pathlib.Path(filename)
+    if not filepath.is_file():
+        logger.error(f"File {filepath} not found")
+        return False
+    start = time.perf_counter()
+    try:
+        with mrcfile.open(filepath) as mrc:
+            data = mrc.data
+    except ValueError:
+        logger.error(
+            f"File {filepath} could not be opened. It may be corrupted or not in mrc format"
+        )
+        return False
+    outfile = filepath.with_suffix(".jpeg")
+    if len(data.shape) != 3:
+        logger.error(
+            f"File {filepath} is not 3-dimensional. Cannot extract central slice"
+        )
+        return False
+    elif len(data.shape) == 3:
```

Comment:
Shouldn't need this line. If it fails the check above then the function returns you would never reach the code below